.github/workflows/test.yml

```diff
@@ -0,0 +1,151 @@
+name: Build & Test
```

Comment:
Can we have a more descriptive name? Also, can we rename the file to maybe `ci.yml`?